Gruntfile.js

```diff
@@ -1036,18 +893,50 @@ module.exports = function(grunt) {
   }
 
   // Register a task, making sure subtasks only run once.
-  function registerTask(grunt, taskName, subTasks) {
-    return grunt.registerTask(taskName, subTasks.map(makeRunOnce));
+  // Description is optional.
+  function registerTask(grunt, taskName, description, subTasks) {
+    if (!subTasks) {
+      subTasks = description;
+      return grunt.registerTask(taskName, subTasks.map(makeRunOnce));
+    }
+    return grunt.registerTask(taskName, description, subTasks.map(makeRunOnce));
   }
 
+  registerTask(grunt, 'default', ['build']);
+
+  // Builds all code, including the "dist" build, but skips
+  // iOS and Android as well as
+  // ts-linting and testing which can be annoying and slow.
+  // We added jshint here because catches hard syntax errors, etc.
+  registerTask(grunt, 'build', [
+    'build_chrome', 
+    'build_firefox', 
+    'build_cca', 
+    'jshint', 
+    'copy:dist', 
+    'jpm:xpi'
+  ]);
+
+  // This is run prior to releasing uProxy and, in addition to
+  // building, tests and lints all code.
+  registerTask(grunt, 'dist', [
+    'build',
+    'test'
+  ]);
+
+  registerTask(grunt, 'compileTypescript', 'Compiles all the Typescript code', [
```

Comment:
I'd prefer to be explicit about it, specially since there's some compilation on the Android side.

On the capitalization, which would you use in an environment that uses underscore: `typescript` or `type_script`? If the first, I'd argue that we should use `Typescript` in CamelCase.